theano/tensor/elemwise.py

```diff
@@ -477,15 +476,17 @@ class Elemwise(OpenMPOp):
 
     """
 
-    def __init__(self, scalar_op, inplace_pattern=None, name=None,
+    __props__ = ("scalar_op", "inplace_pattern", "nfunc_spec", "openmp")
+
+    def __init__(self, scalar_op, inplace_pattern=None,
                  nfunc_spec=None, openmp=None):
         if inplace_pattern is None:
-            inplace_pattern = {}
-        self.name = name
+            inplace_pattern = frozendict({})
         self.scalar_op = scalar_op
-        self.inplace_pattern = inplace_pattern
-        self.destroy_map = dict((o, [i]) for o, i in inplace_pattern.items())
-
+        self.inplace_pattern = frozendict(inplace_pattern)
+        self.destroy_map = dict((o, [i]) for o, i in frozendict(inplace_pattern).items())
+        if not bool(self.inplace_pattern):
+            self.inplace = False
```

Comment:
True. I added this before you had given me a solution to tackle `inplace` and `no_inplace` inside `__str__`